### PR TITLE
Rule S2699: Document false positive

### PR DESCRIPTION
**Pull request by costin-zaharia-sonarsource**
Related to #3751

### CODE REVIEW

File: sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/TestMethodShouldContainAssertion.NUnit.cs [hunk 1]
```diff
@@ -282,6 +282,21 @@ public void TestCaseSource14()
...
+        {
+          new TestCaseData("foo", "bar").Returns(false),
+          new TestCaseData(100, 100).Returns(true),
+        };
+
+        // See: https://github.com/SonarSource/sonar-dotnet/issues/3751
+        // Returned value is verified by the framework and no assertion is needed.
+        [Test]
+        [TestCaseSource(nameof(Cases))] 
+        public bool Test(object obj1, object obj2) // Noncompliant - FP
```
Comment (pavel-mikula-sonarsource) [inline sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/TestMethodShouldContainAssertion.NUnit.cs:296]: Is there a documentation for the `bool` return type? I've only found `void` examples in the [TestCaseSource docs](https://docs.nunit.org/articles/nunit/writing-tests/attributes/testcasesource.html)

File: sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/TestMethodShouldContainAssertion.NUnit.cs [hunk 2]
```diff
@@ -282,6 +282,21 @@ public void TestCaseSource14()
...
+        {
```
Comment (pavel-mikula-sonarsource) [inline sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/TestMethodShouldContainAssertion.NUnit.cs:297]: Can you please add two more testcases with `void` return type? One TP without an assert. One compliant with assert. Just to be sure we don't ignore all methods with that attribute.

**Review by pavel-mikula-sonarsource — CHANGES_REQUESTED** [review]

File: sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/TestMethodShouldContainAssertion.NUnit.cs (hunk 1 above)
Comment (costin-zaharia-sonarsource) [reply sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/TestMethodShouldContainAssertion.NUnit.cs:296]: Not sure why it does not appear anymore in docs, I've used that in the past and the method is still there in  the latest NUnit version (3.12). 

You can find this mentioned in [2.6.4 docs](https://docs.nunit.org/2.6.4/testCaseSource.html)

File: sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/TestMethodShouldContainAssertion.NUnit.cs (hunk 2 above)
Comment (costin-zaharia-sonarsource) [reply sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/TestMethodShouldContainAssertion.NUnit.cs:297]: As far as I can tell both these cases were already covered in the test cases above. See lines 210 and 217.

**Review by pavel-mikula-sonarsource — APPROVED** [review]
LGTM

**Pull request merged**